test/Microsoft.AspNet.Mvc.Core.Test/PhysicalFilePathResultTest.cs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.AspNet.Http.Internal;
+using Microsoft.AspNet.Routing;
+using Microsoft.Net.Http.Headers;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class PhysicalPhysicalFilePathResultTest
+    {
+        [Fact]
+        public void Constructor_SetsFileName()
+        {
+            // Arrange & Act
+            var path = Path.GetFullPath("helllo.txt");
+            var result = new PhysicalFilePathResult(path, "text/plain");
+
+            // Act & Assert
+            Assert.Equal(path, result.FileName);
+        }
+
+        [Fact]
+        public async Task ExecuteResultAsync_FallsbackToStreamCopy_IfNoIHttpSendFilePresent()
+        {
+            // Arrange
+            var path = Path.GetFullPath(Path.Combine("TestFiles", "FilePathResultTestFile.txt"));
+            var result = new PhysicalFilePathResult(path, "text/plain");
```

Comment:
Could we make it so that `PhysicalFilePathResult` doesn't go to disk? Should be able to get away by adding a  `protected virtual Stream GetFileStream(string filePath)` method to the type?